sdks/python/apache_beam/transforms/util.py

```diff
@@ -541,15 +541,15 @@ def process(self, element):
     self._running_batch_size += self._element_size_fn(element)
     if self._running_batch_size >= self._target_batch_size:
       with self._batch_size_estimator.record_time(self._running_batch_size):
-        yield self._batch
+        yield window.GlobalWindows.windowed_value_at_end_of_window(self._batch)
```

Comment:
This makes it consistent, but do you think that it will cause issues?
By doing this we're losing insight into the watermark, any produced element from the batch will automatically inherit the end of global window timestamp.
Should we allow configuring this behavior with a TimestampCombiner?

Reply:
We could allow configuration of this with a TimestampCombiner, but I'd consider that more a feature request than a bug fix. This seems, by default, the safest thing to do in the global window. (Even if we preserved the latest timestamp, there's no limit to how far back the earliest dropped timestamp might be.)

Reply:
Agree that feels like a new feature, I filed #22853